experimental/dds/attributable-map/src/mapKernel.ts

```diff
@@ -386,29 +386,23 @@ export class AttributableMapKernel {
 	}
 
 	private setAttribution(key: string, message?: ISequencedDocumentMessage): void {
-		if (this.options?.attribution?.track) {
-			if (message) {
-				this.attribution?.set(key, { type: "op", seq: message.sequenceNumber });
+		if (message) {
+			this.attribution?.set(key, { type: "op", seq: message.sequenceNumber });
+		} else {
+			if (this.isAttached()) {
+				this.attribution?.set(key, { type: "local" });
 			} else {
-				if (this.isAttached()) {
-					this.attribution?.set(key, { type: "local" });
-				} else {
-					this.attribution?.set(key, { type: "detached", id: 0 });
-				}
+				this.attribution?.set(key, { type: "detached", id: 0 });
 			}
 		}
 	}
 
 	private deleteAttribution(key: string): void {
-		if (this.options?.attribution?.track) {
-			this.attribution?.delete(key);
-		}
+		this.attribution?.delete(key);
```

Comment:
Now that the `options` are gone, this helper becomes redundant. Consider inlining the `this.attribution.delete(key)` (and `this.attribution.clear()`) calls into their callsites.